tests/common/dualtor/dual_tor_io.py

```diff
@@ -180,9 +181,10 @@ def generate_from_t1_to_server(self):
         if self.tor_vlan_intf:
             # If destination VLAN intf is specified,
             # use only the connected server
-            server_ip_list = [self.ptf_intf_to_server_ip_map[
-                self.tor_vlan_intf
-                ]]
+            ptf_port = self.tor_to_ptf_intf_map[self.tor_vlan_intf]
+            server_ip_list = [
+                self.ptf_intf_to_server_ip_map[ptf_port]
```

Comment:
Why was this changed to `ptf_port`? Can it not be kept to the instance variable `self.tor_vlan_intf`?

Reply:
Looks like there was an error in my original code, `self.tor_vlan_intf` should never exist in `self.ptf_intf_to_server_ip_map`. Good catch @lolyu

Reply:
Oh! So I guess the previous code did not hit an error because `self.ptf_intf_to_server_ip_map[self.tor_vlan_intf]]` still returned some IPs based on ptf_port that was same as tor_vlan_intf.

However, this should have failed/caught in link down testcases, where a specific `tor_vlan_intf` is specified. Are we missing any other issue?

Reply:
The link down test cases take down all interfaces, not just a single one. So I don't believe this portion of code was ever run, since all the test cases so far applied any disruptive actions to all ports. There was a similar error in `generate_server_to_t1` but it looks like @lolyu has already fixed it below